test(functional-tests): add minimum coverage PayPal test

Trinaa opened the pull request:
## Because

- We want to maximize test coverage while respecting third party interface boundaries

## This pull request

- adds a test that assures when users click on the PayPal payment option they are directed to a PayPal popup

N/A (Relates To #FXA-9180)

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

Example error message:
<img width="409" alt="Screenshot 2024-03-07 at 10 39 10 AM" src="https://github.com/mozilla/fxa/assets/3422688/1b0341a7-a07d-4484-a17b-736b894e9044">

## Other information (Optional)

Any other information that is important to this pull request.

- Related PR: https://github.com/mozilla/fxa/pull/16477

ashrivastava-qa reviewed (changes requested):
Thanks @Trinaa for the PR. I think looking back at [(https://github.com/mozilla/fxa/pull/16477)] , we removed 4 tests because of the paypal sandbox integration. With this PR we should be adding all those tests with your changes I suppose.

Trinaa commented:
> Thanks @Trinaa for the PR. I think looking back at (#16477) , we removed 4 tests because of the paypal sandbox integration. With this PR we should be adding all those tests with your changes I suppose.

The cause of flake for those four tests occurred because we were waiting for and interacting with the PayPal checkout form. See [example](https://app.circleci.com/pipelines/github/mozilla/fxa/47889/workflows/4525cb21-7aef-44e4-ae32-cefcdeece1f0/jobs/485257/tests) 
<img width="590" alt="image" src="https://github.com/mozilla/fxa/assets/3422688/afc7cfa5-5b69-4695-abf5-3467f527d0aa">

I've done nothing in this pull request that I believe would address this issue, so I don't think we can add the cases back. 

I'm not certain that over time this case I've added will not also be intermittent, but we don't have evidence of flake before this point at this time and we can at least check that the button works as expected.

Trinaa commented:
Clarification from 2023/03/11 ETE Meeting <-- Add a test to cover updating subscription with PayPal button, using the credit card steps as per-requisite.

Trinaa commented:
> **Clarification from 2023/03/11 ETE Check-In for FxA & SubPlat** Add tests for the `resubscription` and `update mode of payment` use cases using the stripe/credit card flow as prerequisite steps.

@ashrivastava-qa This could be due to my inexperience, but I'm not sure it's possible to add these test cases as you suggest. I don't see PayPal as an option in the `resubscruption` or `update mode of payments` flows if they are initiated with credit card.

example:
<img width="695" alt="image" src="https://github.com/mozilla/fxa/assets/3422688/29600d7a-371e-405c-9882-f33677487df4">

Could you clarify the steps you meant?

ashrivastava-qa commented:
oh! I see what you mean. I don't remember if this is how it was set up since beginning or if this is a new development with 'update mode of payment' to not allowing switching between mode of payments. I think in this case we should just go ahead with you initial PR. Thanks for working on it! :)

ashrivastava-qa reviewed (approved):
LGTM :)
